service/src/main/java/uk/gov/hmcts/reform/fpl/service/ChildrenMapperService.java

```diff
@@ -0,0 +1,34 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fpl.model.Children;
+
+import java.util.Map;
+import java.util.Optional;
+
+@Service
+public class ChildrenMapperService {
+
+    private final ObjectMapper mapper;
+    private final Logger logger = LoggerFactory.getLogger(ChildrenMapperService.class);
+
+    @Autowired
+    public ChildrenMapperService(ObjectMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    public Optional<Children> mapChildren(final Map<String, Object> childrenMap) {
```

Comment:
There should be one generic mapping services used to deserialise any data structure.